table.cc

```diff
@@ -480,6 +474,48 @@ table::make_reader(schema_ptr s,
     }
 }
 
+flat_mutation_reader
+table::make_reader(schema_ptr s,
+                           const dht::partition_range& range,
+                           const query::partition_slice& slice,
+                           const io_priority_class& pc,
+                           tracing::trace_state_ptr trace_state,
+                           streamed_mutation::forwarding fwd,
+                           mutation_reader::forwarding fwd_mr) const {
+    auto* semaphore = _config.read_concurrency_semaphore;
+
+    auto ms = mutation_source([this] (
+            schema_ptr s,
+            const dht::partition_range& pr,
+            const query::partition_slice& slice,
+            const io_priority_class& pc,
+            tracing::trace_state_ptr trace_state,
+            streamed_mutation::forwarding fwd,
+            mutation_reader::forwarding fwd_mr,
+            reader_resource_tracker tracker) {
+        return do_make_reader(std::move(s), pr, slice, pc, std::move(trace_state), fwd, fwd_mr, tracker);
+    });
+    return maybe_restrict_reader(semaphore, std::move(s), range, slice, pc, std::move(trace_state),
```

Comment:
nit: you could pass `_config.read_concurrency_semaphore` here directly.

Reply:
yes